### PR TITLE
removed validations from serial number

### PR DESCRIPTION
**Pull request by oliveromahony**
### Proposed changes
Remove the issuer validations

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [ ] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [ ] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: test/helpers/cert_utils.go [hunk 1]
```diff
@@ -44,14 +45,17 @@ func GenerateSelfSignedCert(t testing.TB) (keyBytes, certBytes []byte) {
 	keyBytes = x509.MarshalPKCS1PrivateKey(key)
 
 	tmpl := x509.Certificate{
-		NotBefore:    time.Now(),
+		NotBefore:    time.Now().Add(-10 * time.Minute), // Allow for clock skew
```
Comment (dhurley) [inline test/helpers/cert_utils.go:48]: Just curious why 10 minutes needs to be removed from the time

File: internal/collector/otel_collector_plugin.go [hunk 2]
```diff
@@ -43,6 +43,8 @@ var _ bus.Plugin = (*Collector)(nil)
 
 // NewCollector is the constructor for the Collector plugin.
 func New(conf *config.Config) (*Collector, error) {
+	initMutex := &sync.Mutex{}
```
Comment (dhurley) [inline internal/collector/otel_collector_plugin.go:46]: Is this mutex necessary?

**Review by dhurley — APPROVED** [review]

File: pkg/files/file_helpers_test.go [hunk 3]
```diff
@@ -227,3 +230,48 @@ func TestConvertX509SignatureAlgorithm(t *testing.T) {
 		})
 	}
 }
+
+func TestMarshalAndUnmarshal(t *testing.T) {
```
Comment (aphralG) [inline pkg/files/file_helpers_test.go:234]: is this needed ?

File: test/helpers/cert_utils.go (hunk 1 above)
Comment (oliveromahony) [reply test/helpers/cert_utils.go:48]: just incase there is clock drift

Comment (oliveromahony) [reply test/helpers/cert_utils.go:48]: I removed it

**Review by dhurley — APPROVED** [review]

**Review by aphralG — APPROVED** [review]

**Pull request merged**